lib/bolt/project_migrate/base.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'bolt/error'
+
+module Bolt
+  class ProjectMigrate
+    class Base
+      def initialize(outputter)
+        @outputter = outputter
+      end
+
+      protected def print_step(step)
+        first, *remaining = wrap(step).lines
+        first = indent("→ #{first}", 2)
+        remaining = remaining.map { |line| indent(line, 4) }
+        step      = [first, *remaining, "\n"].join("\n")
+        @outputter.print_message(step)
+      end
+
+      protected def wrap(string)
+        string.gsub(/(.{1,76})(\s+|\Z)/, "\\1\n")
+      end
+
+      protected def indent(string, size = 2)
+        indent = "\s" * size
+        string.gsub(/^/, indent.to_s)
+      end
+
+      protected def backup_file(origin_path, backup_dir)
```

Comment:
This could potentially go in the Project class if we wanted to get rid of this base class altogether, but I'm also fine with leaving it here.